pennylane/gradients/hadamard_gradient.py

```diff
@@ -94,6 +96,7 @@ def hadamard_grad(
     argnum=None,
     aux_wire=None,
     device_wires=None,
+    mode: str = "standard",
```

Comment:
```suggestion
    mode: Literal["standard", "reversed", "direct", "reversed-direct"] = "standard",
```
with a `from typing import Literal` up top.

Strictly optional, but can be nice for type checking. It more clearly communicates what string values are allowed.

Reply:
I'll vote for it since there's quite some chance near future we need to use them frequently. A clear typehint may help a lot of devs

Reply:
updated the signature, added import and test written